src/components/shows_container/ShowsContainer.js

```diff
@@ -1,38 +1,39 @@
-import React, { useEffect, useContext } from 'react';
+import React, { useEffect, useContext, useCallback, useState } from 'react';
 import Show from '../show_card/ShowCard';
 import { ShowContext } from '../../contexts/ShowContext';
 import phishLogoLoading from '../../phishLogoLoading.png';
 import './ShowsContainer.css';
 
 const ShowsContainer = ({ year }) => {
   const { shows, getShows } = useContext(ShowContext);
-  // const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    getShows(year);
-  }, [getShows, year]);
+  const memo = useCallback(() => {
+    const renderShows = async () => {
+      await getShows(year);
+      setIsLoading(false)
+    }
+    renderShows()
```

Comment:
Can we set the loading to false after we hit renderShows()? Just if this makes for a smoother transition for the fish loading page to the displayed shows.

Reply:
I believe essentially that would be the same functionality we have currently ... Essentially if I had the code read renderShows.then(()=> setIsLoading(false) would be the same things since we have await there.
  
const memo = useCallback(() => {
    const renderShows = async () => {
      await getShows(year);
      setIsLoading(false). ***
    }
    renderShows()
  }, [year, getShows])


render is maybe not a good name for this function though. I will change that. Because essentially the only thing it is doing is fetching the shows and changing our loading state.